### PR TITLE
Mention HTTP_PROXY if download error occurs; fall back to NPM's proxy…

### LINKED ISSUE #1662: When the binary fails to download add a message about proxy
In CLI if the binary download fails, in the error message tell about proxy environment variables.

### LINKED ISSUE #4719: Add fallback to NPM proxy configuration when other proxy configuration is unavailable
Add support for the `proxy` and `https-proxy` configuration variables in NPM to control Cypress's proxy configuration.

Users should still be able to do `HTTP_PROXY=` (undefined) to disable proxy usage altogether.

This should allow Cypress to "just work" when launched by proxy users who use NPM as a launcher.

This should also be added to the CLI so that it can use `npm_config_proxy` at `npm install` time.

### PR DESCRIPTION
**Pull request by flotwig**
- Fixes #1662
- Fixes #4719
- Updates download error message
- Adds fallback to npm_config_proxy, npm_config_https_proxy

### CODE REVIEW

File: cli/lib/tasks/download.js [hunk 1]
```diff
@@ -180,6 +190,7 @@ const downloadFromUrl = ({ url, downloadDestination, progress }) => {
 
     const req = request({
       url,
+      proxy: getProxyUrl(),
```
Comment (bahmutov) [inline cli/lib/tasks/download.js]: proxy is one of those parameters that we should log above, just like we are `debug` logging `url`.

**Comment by brian-mann** [comment]
This will download the binary but still fail when running actual Cypress tests because we don't check those same npm config env vars right?

**Comment by brian-mann** [comment]
Should we update the docs to make mention of us respecting the `npm` config env vars?

**Comment by flotwig** [comment]
> This will download the binary but still fail when running actual Cypress tests because we don't check those same npm config env vars right?

Correct, I think it makes sense to respect the npm config vars here because it's part of the `npm install` when they download the package. 

I figure that Cypress might not be launched through NPM, so npm_config might not be available, so we should only accept HTTP_PROXY in the server for it to behave uniformly. Not sure, do you think it'd be best to allow npm_config_proxy as a fallback in the server as well?

> Should we update the docs to make mention of us respecting the `npm` config env vars?

The `npm install` will fail before reaching Cypress if they don't have their proxy configured properly, so I don't think it's necessary to mention in *our* docs. They'll have to configure `npm config proxy` anyways before Cypress comes into the picture.

**Comment by brian-mann** [comment]
@bahmutov 	I don't think so - proxies aren't really something that users **desire** to have setup - they are constraints that exist in their work environment that they have to conform to. It's not like "a feature" or anything.

As long as we provide good documentation / error messaging, etc that help guide and direct those who are affected I think we're fine. Have we had anyone open issues or ask support questions in regards to how to utilize their proxies? I'm not aware of any TBH.

**Comment by brian-mann** [comment]
@flotwig 	this PR is actually doing two things - adding a message but also tapping into the npm config for proxy support. Can you open a separate issue for that (since it's a discrete thing that needs to be mentioned in the changelog and also noted in the docs).

**Comment by flotwig** [comment]
> Can you open a separate issue for that (since it's a discrete thing that needs to be mentioned in the changelog and also noted in the docs).

Done, #4719

**Pull request merged**